ci-tools/release/tools/rtl_hash.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+# Licensed under the Apache-2.0 license
+
+# This tool is used to generate a hash over the Caliptra RTL files
+# This can be used to verify the integrity of the RTL after it has been integrated
+# The hash for the release is included in the release_notes.txt
+# 
+# Usage: rtl_hash.sh <path_to_rtl_src_dir> <rtl_file_list>
+#   path_to_rtl_src_dir     Path to the Caliptra RTL files. This is the src/ directory within 
+#                           the caliptra RTL repo. This may differ once integrated
+#   rtl_file_list           This list of all the files that should be included in the hash
+#                           is generated and packaged with the release (rtl_hash_file_list.txt)
+#
+# Certain files are expected to be modified during integration. These are mentioned in
+# the CaliptraIntegrationSpecification.md in the RTL repo and are excluded from the rtl hash
+# file list used to generate the RTL hash
+#
+# If any files in the list are not found, an error will be output and mention all missing files
+# The hash will not be computed since it will not match the release hash with any files missing
+
+# Exit and report failure if anything fails
+set -euo pipefail
+
+# Check arg count
+if [ $# -ne 2 ]
+  then
+    echo "Usage: $(basename $0) <path_to_rtl_src_dir> <rtl_file_list>"
+	exit -1
+fi
+
+# Get args
+rtl_path=$1
+expected_file_list=$2
+
+# Read expected file list
+expected_file_list=$(cat $expected_file_list)
+# Prepend rtl path
+expected_file_list=$(echo "$expected_file_list" | sed "s@^@"$rtl_path"/@")
+
+# Make sure all files exist
+missing_files=0
+while read line; do
+	# Check if the file is missing
+	if ! test -f $line; then
+		# Report any missing files (and keep count)
+		if [ "$missing_files" -eq 0 ]; then
+			echo "Missing expected files: "
+		fi
+		missing_files=$(($missing_files + 1))
+		echo "  $line"
+	fi
+done < <(echo "$expected_file_list")
+
+# Calculate the hash (only if no files were missing)
+if [ "$missing_files" -eq 0 ]; then
+	hash=$(cat $expected_file_list | sha1sum | tr -d "\n *-")
```

Comment:
sha1 is compromised. sha256sum or sha384sum?

Reply:
Honestly the main reason I used this was because it was shorter and matched what git uses for the hashes listed above it in the notes. My understanding was sha1 was okay for non-security based uses like this where it's just necessary to identify something uniquely. But I can see the argument that this tool may have some security implications if we don't trust everyone with access to the RTL during integration.

Reply:
I would not consider this a security feature. It is already going to have to be taken on trust and faith that an integrator is truly using the RTL they claim to be using.

Reply:
That was my original thinking. I'm open to either way on this though. I like that it's more concise and consistent with git with sha1. But I can switch it to sha256 if you feel more comfortable with it, Kor.

Reply:
I don't have a strong opinion, other than I know sometimes having sha1 deps can cause pain with NIST.

Reply:
(As soon as github supports sha256 I would love to migrate the repo)

Reply:
This is updated to sha384 now